src/ball/mod.rs

```diff
@@ -53,9 +53,11 @@ impl Ball {
         let ball = self.position;
 
         players.iter().map(|p| p.paddle_vertex()).for_each(|p| {
-            if p.chunks(2)
-                .any(|c| dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS)
-            {
+            if p.chunks(2).any(|c| {
+                dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS
+                    && ((self.position[0] < WINDOW_WIDTH / 2.0 && self.direction[0] < 0.0)
```

Comment:
What do you think of adding a variable for these checks.

Something like:

```rust
let is_ball_vertial_stuck = (self.position[0] < WINDOW_WIDTH / 2.0 && self.direction[0] < 0.0)
            || (self.position[0] > WINDOW_WIDTH / 2.0 && self.direction[0] > 0.0);
if p.chunks(2).any(|c| {
    dist_to_segment(ball, c[0], c[1]) < BALL_RADIUS
        && is_ball_vertial_stuck
}) {
    self.direction[0] = -self.direction[0];
}
```

The name is just an example, if you have something better go for it :slightly_smiling_face: